### PR TITLE
Seurat 3.1

### PR DESCRIPTION
**Pull request by astrovsky01**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)
Recreating previous seurat script in 3.1, adding new function options and replacing the current rscript output with a knitted Rscript file

### CODE REVIEW

**Comment by nsoranzo** [comment]
@astrovsky01 I'm interested in this, please ping me when it's ready for review or if you need any help!

**Comment by astrovsky01** [comment]
@pcm32 I have (mostly) updated the full workflow from the original single seurat tool that perfomed the full workflow and output the pdf. Modifying that into a knitted pdf, I was then planning on adding in the new tools on top as additional selections for functions to run. If you have wrapped the new functions, I think merging the efforts would be a great idea, and I'll push what I've done so far now.

**Comment by astrovsky01** [comment]
Current status: output pdf is entirely customizable. You can choose which graphs you would like to show up, whether or not you want code next to them, etc.

**Comment by nsoranzo** [comment]
The error now is `sh: 1: pdflatex: not found`, which should be solved by adding `<requirement type="package" version="0.1.11">tectonic</requirement>`

File: tools/seurat/seurat.xml [hunk 1]
```diff
@@ -1,110 +1,110 @@
...
+        warn = \"${meta.warn}\",
+        varstate = \"${meta.varstate}\",
+        showcode = \"${meta.showcode}\",
+        vlnfeat = \"$vln\",
+        featplot = \"$feat\",
+        PCplots = \"$PCs\",
+        tsne = \"$tsne\",
+        heatmaps = \"$heatmaps\"),
+    intermediates_dir = \".\",
+    output_format = pdf_document(),
```
Comment (nsoranzo) [inline tools/seurat/seurat.xml]:
```suggestion
    output_format = pdf_document(latex_engine=\"tectonic\"),
```
was what I was hoping could work, unfortunately it fails with:
```
Error in match.arg(latex_engine, c("pdflatex", "lualatex", "xelatex")) :
'arg' should be one of \u201cpdflatex\u201d, \u201clualatex\u201d, \u201cxelatex\u201d
Calls: render -> pdf_document -> match.arg
Execution halted
```
due to https://github.com/rstudio/rmarkdown/blob/master/R/pdf_document.R#L147 

So, we either need a working pdflatex in conda-forge or extend rmarkdown to support tectonic.

**Comment by pcm32** [comment]
@astrovsky01 is `as.loom` call working for you with conda package r-seurat 3.1.0 that you are using here as dependency? Sorry that I ask this here, but loomR repo seems rather abandoned. Or are you writing to loom through some other mechanism? Thanks

**Comment by astrovsky01** [comment]
@pcm32 it is not. I keep getting format errors when using files I know for a fact read in properly in a python script with loompy.

**Comment by bgruening** [comment]
ping @astrovsky01

**Comment by astrovsky01** [comment]
I have no clue how to deal with the latex issue...

File: tools/seurat/seurat.xml [hunk 2]
```diff
@@ -1,111 +1,109 @@
...
+            <param name="logfc_threshold" type="float" min="0" value="0.25" label="LogFC threshold"
                 help="With FindMarkers, limit testing to genes which show, on average, at least X-fold difference (log-scale)between the two groups of cells. Default is 0.25 Increasing logfc.threshold speeds up the function, but can miss weaker signals." />
         </section>
+        <section name="meta" title="Output options" expanded="true">
+            <param name="showcode" type="boolean" truevalue="T" falsevalue="F" checked="false" label="Show code alongside outputs?"/>
+            <param name="warn" type="boolean" truevalue="T" falsevalue="F" checked="false" label="Include warnings in the output file (Yes) or pipe to stdout (No)"/>
+            <param name="varstate" type="boolean" truevalue="T" falsevalue="F" checked="false" label='Display variable values used in code at the beginning of output file                                                                                     ?'/>
```
Comment (nsoranzo) [inline tools/seurat/seurat.xml]:
```suggestion
            <param name="varstate" type="boolean" truevalue="T" falsevalue="F" checked="false" label="Display variable values used in code at the beginning of output file?"/>
```

**Comment by astrovsky01** [comment]
ping @bgruening

**Comment by bgruening** [comment]
@astrovsky01 this is not WIP anymore?

**Comment by astrovsky01** [comment]
Since @nsoranzo helped out, it's all good. Next update will add in hd5 formats, but as discussed above, Seurat is having a bit of an issue with that right now.

**Comment by bgruening** [comment]
@astrovsky01 can you remove the large HTML and use asserts? The asserts might be a little bit complex to also check for the embedded images ... but maybe its possible?

@mblue9 can you maybe check the R code?

Thanks!

**Comment by astrovsky01** [comment]
I just made the test file much smaller and kept sim_size

**Comment by nsoranzo** [comment]
The last error is due to a Galaxy bug, which should be fixed by https://github.com/galaxyproject/galaxy/pull/9067

**Comment by nsoranzo** [comment]
Green!

**Pull request merged**